src/servers/src/error.rs

```diff
@@ -737,8 +737,22 @@ impl From<std::io::Error> for Error {
     }
 }
 
+fn log_error_if_necessary(error: &Error) {
+    if error.status_code().should_log_error() {
+        error!(error; "Failed to handle HTTP request: ");
+    } else {
+        debug!("Failed to handle HTTP request: {error:?}");
```

Comment:
Can we change it to info? We still need to know what the error is.
```suggestion
        info!("Failed to handle HTTP request: {error:?}");
```